identity-account/src/types/encryption_key.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use identity_account_storage::storage::Storage;
+use identity_account_storage::types::KeyLocation;
+use identity_core::crypto::PublicKey;
+use identity_did::verification::MethodScope;
+use identity_iota_core::did::IotaDID;
+use identity_iota_core::document::IotaDocument;
+use identity_iota_core::document::IotaVerificationMethod;
+
+use crate::Error;
+use crate::Result;
+
+/// Supported keys for encrypting data
+#[derive(Clone, Debug)]
+pub enum EncryptionKey {
+  Ed25519,
+  X25519(PublicKey),
+}
```

Comment:
I don't think `EncryptionKey` is necessary, it can usually be inferred from the verification method referenced by `fragment` and the public key can be part of a rich enum specifying with encryption algorithm to use.